### PR TITLE
[DEVX-265] : Improve README.md of Python SDK

### PR DESCRIPTION
**Pull request by adithyan-sukumar**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -25,6 +25,31 @@ This is the official Python client for interacting with our powerful [API](https
 
 ---
 
+
+
+##Table Of Contents
```
Comment (isaac-chung) [inline README.md:30]:
```suggestion
## Table Of Contents
```

File: README.md [hunk 2]
```diff
@@ -25,6 +25,31 @@ This is the official Python client for interacting with our powerful [API](https
...
+
+**[1. Installation](#installation)**
+**[2. Getting Started](#getting-started)**
+**[3. Interacting with Datasets](#interacting-with-datasets)**
+**[4. Interacting with Inputs](#interacting-with-inputs)**
```
Comment (isaac-chung) [inline README.md:35]: Need a new line in between. Right now all 4 are on the same line.

File: README.md [hunk 3]
```diff
@@ -49,8 +74,17 @@ pip3 install -r requirements.txt
...
+* ***Log into Portal &rarr; Profile Icon &rarr; Security Settings &rarr; Create Personal Access Token &rarr; Set the scopes &rarr; Confirm***
+
+
 Export your PAT as an environment variable. Then, import and initialize the API Client.
 
+[Get User ID:](https://help.clarifai.com/hc/en-us/articles/4408131912727-How-do-I-find-my-user-id-app-id-and-PAT-)
+
+* ***Log into Portal &rarr; Profile Icon &rarr; Account &rarr; Profile &rarr; User-ID***
+
+Set PAT as environment variable through terminal:
```
Comment (isaac-chung) [inline README.md:87]: Maybe make "Create PAT" as a bullet point, and put the description on the same line? 
Same for User ID. 
Then, group both bullets together above the line "Export your PAT...".

**Review by isaac-chung — COMMENTED** [review]
Great start! Added a few comments.

File: README.md (hunk 2 above)
Comment (adithyan-sukumar) [reply README.md:35]: image

They are on separate lines when previewed.

File: README.md (hunk 3 above)
Comment (adithyan-sukumar) [reply README.md:87]: image

Is this fine?

Comment (isaac-chung) [reply README.md:87]: Yes, thanks!

File: README.md (hunk 2 above)
Comment (isaac-chung) [reply README.md:35]: That's odd. Both my preview in VS Code and Github shows that they are on the same line....
Screenshot 2023-10-30 at 11 52 04 AM
^ VS Code

Screenshot 2023-10-30 at 11 52 54 AM


^ Github. [Try click this link to see the preview in Github](https://github.com/Clarifai/clarifai-python/blob/a06ca909a198b7adcbc1da5079e1c2c81680713f/README.md).

Comment (adithyan-sukumar) [reply README.md:35]: OK, I will check that out.

Comment (adithyan-sukumar) [reply README.md]: I have made the changes. Can you check?

File: README.md [hunk 4]
```diff
@@ -74,16 +74,13 @@ pip3 install -r requirements.txt
 ## Getting started
 Clarifai uses **Personal Access Tokens(PATs)** to validate requests. You can create and manage PATs under your Clarifai account security settings.
 
-[Create PAT:](https://docs.clarifai.com/clarifai-basics/authentication/personal-access-tokens/)
-* ***Log into Portal &rarr; Profile Icon &rarr; Security Settings &rarr; Create Personal Access Token &rarr; Set the scopes &rarr; Confirm***
+* [Create PAT:](https://docs.clarifai.com/clarifai-basics/authentication/personal-access-tokens/) ***Log into Portal &rarr; Profile Icon &rarr; Security Settings &rarr; Create Personal Access Token &rarr; Set the scopes &rarr; Confirm***
+<br />
```
Comment (isaac-chung) [inline README.md:78]: Screenshot 2023-10-30 at 1 16 06 PM
Maybe remove this <br>? Looks a bit odd to have that much space between 2 bullets.

Comment (isaac-chung) [reply README.md:78]: Otherwise, LGTM

Comment (adithyan-sukumar) [reply README.md:78]: ok, I will change it.

Comment (isaac-chung) [reply README.md]: Thanks! Looks good, but it seems like some linting tests are failing. Did you set up your pre-commit hooks?

Comment (adithyan-sukumar) [reply README.md]: image

yes.

**Comment by sainivedh** [comment]
Could we improve the code snippets portion on README, now its just dump of the code. It would be helpful to make it very verbose and maybe very interactive/intuitive.

File: README.md [hunk 5]
```diff
@@ -25,6 +25,31 @@ This is the official Python client for interacting with our powerful [API](https
...
 
+
+
+## Table Of Contents
+
+**[1. Installation](#installation)** \
+**[2. Getting Started](#getting-started)** \
+**[3. Interacting with Datasets](#interacting-with-datasets)** \
+**[4. Interacting with Inputs](#interacting-with-inputs)** 
+* [4.1 Input upload](#input-upload)
```
Comment (sainivedh) [inline README.md:36]:
```suggestion
* [4.1 Input Upload](#input-upload)
```

File: README.md [hunk 6]
```diff
@@ -25,6 +25,31 @@ This is the official Python client for interacting with our powerful [API](https
...
+* [4.2 Input listing](#input-listing)
```
Comment (sainivedh) [inline README.md:37]:
```suggestion
* [4.2 Inputs Listing](#input-listing)
```

File: README.md [hunk 7]
```diff
@@ -25,6 +25,31 @@ This is the official Python client for interacting with our powerful [API](https
 
 ---
 
+
+
+## Table Of Contents
+
+**[1. Installation](#installation)** \
```
Comment (sainivedh) [inline README.md:32]: Either of numbering/bullet point should be good. I'm not sure if we need both. Could we use just bullet points

**Review by sainivedh — COMMENTED** [review]
Good work. Added some comments

Also a minor spelling issue
`capabiities` at the SDK description

File: README.md (hunk 4 above)
Comment (isaac-chung) [reply README.md]: This looks good to me, thanks for iterating. Just need to figure out why the tests are failing and address @sainivedh 's comments :)

File: README.md (hunk 7 above)
Comment (adithyan-sukumar) [reply README.md]: I have made the changes.Can you check.

File: README.md [hunk 8]
```diff
@@ -18,13 +18,38 @@ Clarifai Python SDK</a>
...
+  * [Input Upload](#input-upload)
+  * [Input Listing](#input-listing)
+
+* **[Interacting with Models](#interacting-with-models)**
+  * [Workflow Predict](#workflow-predict)
+  * [Workflow Listing](#workflows-listing)
+  * [Workflow Create](#workflow-create)
+  * [Workflow Export](#workflow-export)
+
+* **[More Examples](#more-examples)**
```
[the remark this answers is not in the document]
Comment (isaac-chung) [reply README.md:45]: I quite like the sub-list @sainivedh . Are you referring to those?

[the remark this answers is not in the document]
Comment (sainivedh) [reply README.md:45]: This regarding the spacing between main bullet points
```suggestion
## Table Of Contents
- **[Installation](#installation)**
- **[Getting Started](#getting-started)**
- **[Interacting with Datasets](#interacting-with-datasets)**
- **[Interacting with Inputs](#interacting-with-inputs)**
  * [Input Upload](#input-upload)
  * [Input Listing](#input-listing)
- **[Interacting with Models](#interacting-with-models)**
  * [Workflow Predict](#workflow-predict)
  * [Workflow Listing](#workflows-listing)
  * [Workflow Create](#workflow-create)
  * [Workflow Export](#workflow-export)
- **[More Examples](#more-examples)**
```
image

This regarding the spacing This regarding the spacing

File: README.md [hunk 9]
```diff
@@ -166,7 +196,8 @@ all_llm_community_models = App().list_models(filter_by={"query": "LLM",
 all_llm_community_models = list(all_llm_community_models)
 ```
 
-### Interacting with Workflows
+### 🔥 Interacting with Workflows
```
Comment (sainivedh) [inline README.md:199]:
```suggestion
## 🔥 Interacting with Workflows
```

File: README.md [hunk 10]
```diff
@@ -125,8 +155,8 @@ all_concepts = list(app.list_concepts())
 ```
 
 
-### Interacting with Models
-
+### 🥁 Interacting with Models
```
Comment (sainivedh) [inline README.md:158]:
```suggestion
## 🥁 Interacting with Models
```

File: README.md [hunk 11]
```diff
@@ -66,7 +94,8 @@ apps_generator = client.list_apps()
 apps = list(apps_generator)
 ```
 
-### Interacting with Datasets
+### 💾 Interacting with Datasets
```
Comment (sainivedh) [inline README.md:97]:
```suggestion
## 💾 Interacting with Datasets
```

File: README.md [hunk 12]
```diff
@@ -91,9 +120,10 @@ Dataset().export(save_path='output.zip', local_archive_path='clarifai-data-proto
 ```
 
 
-### Interacting with Inputs
+### ⌨️ Interacting with Inputs
```
Comment (sainivedh) [inline README.md:123]:
```suggestion
##  💾 Interacting with Inputs
```
Inputs is also a data, so having keyboard icon this seems misleading wr.t python `input()`

File: README.md [hunk 13]
```diff
@@ -166,7 +196,8 @@ all_llm_community_models = App().list_models(filter_by={"query": "LLM",
 all_llm_community_models = list(all_llm_community_models)
 ```
 
-### Interacting with Workflows
+### 🔥 Interacting with Workflows
+Using ***Workflow()***, you can interact with Calrifai workflows in many ways. It allows prediction and modification of inference thresholds to get desirable results. Creating and exporting of workflows can be done using local ***yaml*** file.
```
Comment (sainivedh) [inline README.md:200]:
```suggestion
Workflows offer a versatile framework for constructing the inference pipeline, simplifying the integration of diverse models. You can use the **Workflow** class to create and manage workflows using **YAML** configuration.

For starting or making quick adjustments to existing Clarifai community workflows using an initial YAML configuration, the SDK provides an export feature.
```

File: README.md [hunk 14]
```diff
@@ -125,8 +155,8 @@ all_concepts = list(app.list_concepts())
 ```
 
 
-### Interacting with Models
-
+### 🥁 Interacting with Models
+Using Clarifai models, you can choose any models  by provding model URL or ID to perform prediction and  list all available models. It is also possible to modify model prediction outcomes by adding threshold.
```
Comment (sainivedh) [inline README.md:159]: Descriptions could be made more verbose and detailed.
```suggestion
The **Model** Class allows you to perform predictions using Clarifai models. You can specify which model to use by providing the model URL or ID. This gives you flexibility in choosing models. The **App** Class also allows listing all available Clarifai models for discovery. 

For greater control over model predictions, you can pass in an `output_config` to modify the model output as demonstrated below. 
```

**Review by sainivedh — COMMENTED** [review]
Added some suggestions on sizing and descriptions

**Review by sainivedh — APPROVED** [review]
lgtm👍

**Review by isaac-chung — APPROVED** [review]
Looks good. Thanks for iterating!

**Pull request merged**